code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -86,6 +86,8 @@ namespace <#= qualifiedNamespace #>
                 new ComponentPool<ComponentsUpdated<<#= componentDetails.TypeName #>.Update>>(
                     () => new ComponentsUpdated<<#= componentDetails.TypeName #>.Update>(),
                     (component) => component.Buffer.Clear());
+                    
+            private readonly Dictionary<Unity.Entities.Entity, <#= componentDetails.TypeName #>.Update> possibleUpdates = new Dictionary<Unity.Entities.Entity, <#= componentDetails.TypeName #>.Update>();
```

Comment:
Should these sorts of collections be lazily initialized? Are they guaranteed to be used 100% of the time?
If not, could we use the pattern `private asdf Blah { get => blah ?? ((blah = new asdf() )) }` ?

Reply:
This will always be used, when you receive OnComponentUpdate you don't know whether you will get authority and if you don't you have to add a ComponentUpdated so I always store the value in a dictionary.

Reply:
This would involve generating a system per class and registering it. Do you think this would be fine? Don't think it would be possible to have one global system because it needs to be aware of all of the fields of the component.

Reply:
I wouldn't generate a system for each one, that makes it hard to see what systems are running on a world and makes it hard to talk about what systems are part of core. I'd make one system which does that and make it call out to something generated. It might also be that the benefits don't outweigh the dis-benefits however I suspect that this is worth trying. Might be worth catching up with @jamiebrynes7 about whether or not the core is going to contain an easy to use AddUpdate method of a given component type when the core changes go in.

Reply:
An `AddUpdate` method as currently exists in the `MutableView`? Probably not - will pull that out into generated code as much as possible

Reply:
It would have to go in generated code, but the question is will it be accessible. We'd also need a helper method for turn data to update (which you likely want for users anyway). If you have a single system doing this then it would likely have to iterate over whatever the translation equivalent is and call a function which matches new components and adds an update to the entity.

Reply:
It might be best to just use the book keeping in the current core and do this in the new one. Or just go straight to the new one.

Reply:
I have implemented a version using group filters and added it to the design doc. Please have a look: https://brevi.link/reactive-component-addition